### PR TITLE
Some performance improvements to CWS and the submissions page in particular

### PR DESCRIPTION
**Pull request by stefano-maggiolo**
Test setup:
- 1 contest (coming from functional test)
- 5 tasks
- 1 dataset/task
- 50 submissions/task
- 150 participations
- repeatedly querying the submissions page with 1 or 5 threads 

```
Master:                 4.05 qps (1 thread) /  5.64 qps (5 threads)
Joinedload SR:         11.90                / 13.51
Passing participation: 12.34                / 15.67
Skip user query:       13.33                / 19.08
```

---
This change is [Reviewable](https://reviewable.io/reviews/cms-dev/cms/657)

### CODE REVIEW

File: cms/db/contest.py [hunk 1]
```diff
@@ -542,7 +524,7 @@ def tokens_available(self, username, task_name, timestamp=None):
         future. Also, if r[0] == 0 and r[1] is None, then r[2] should
         be ignored.
 
-        username (string): the username of the user.
+        participation (Participaiton): the participation.
```
Comment (veluca93) [inline cms/db/contest.py]: Participaiton should be Participation

File: cms/db/contest.py [hunk 2]
```diff
@@ -542,7 +524,7 @@ def tokens_available(self, username, task_name, timestamp=None):
...
         task_name (string): the name of the task.
```
Comment (veluca93) [inline cms/db/contest.py]: Since we pass a Participation object, shouldn't we pass a Task object instead of a task_name string?

Comment (stefano-maggiolo) [inline cms/db/contest.py]: Good point.

File: cms/db/contest.py (hunk 1 above)
Comment (stefano-maggiolo) [inline cms/db/contest.py]: Done

**Pull request merged**